### PR TITLE
[Rdy] [CR] Skeletons overhaul

### LINKED ISSUE #23463: Overhaul skeletons to be more in line with the remainder of zombie types
### Expected behaviour
Zombie types resemble each other.
### Actual behaviour
Skeletons are basically fantasy skeletons with a scifi-flavored paint-job.
### Steps to reproduce the behaviour
Look at the mon_skeleton definition.

Skeletons have several qualities:
Bony, HARDTOSHOOT, slashing-based attacks.
Let's keep those, but adjust the description and properties to bring them in line with how we understand the blob to work.

### Description ###
Instead of having no or minimal flesh, have the bones move to an exoskeleton.
In particular, the ribcage fuses into a carapace, organs move into the ribcage, and limb bones enlarge to protect muscles.  Meanwhile muscles become smaller (no free lunch), so they get slower overall.

### Stats ###
Armor goes up, speed goes down, slashing attacks stay the same (hands are replaced with bone spikes?), and HARDTOSHOOT is redefined as "makes creature count as one size smaller for the purpose of shooting" instead of a flat 50% miss chance.

This keeps most of the existing properties of the skeleton, makes them fit in the game world better, and opens a path to making a exoskeleton-based evolution path.  i.e somewhat bony just gives talons and some armor, skeleton has a full carapace, then the size and thickness increase until you have a Skeletal Juggernaut that's 8' tall with 3" thick bone plates covering it.

### PR DESCRIPTION
**Pull request by nexusmrsep**
Resolves #23463 

This may be ~~part one of~~ the skeleton overhaul proposed by @kevingranade .

**Things done in this pull request:** 
- descriptions of skeleton and skeletal dog
- armour increased by 50%
- [edit] speed reduced due to overgrown bone carapace and reduced muscle mass
- "HARDTOSHOOT" flag reduces size of the creature for hit chance calculations (ranged only)
- skeletal  juggernaut

### CODE REVIEW

File: data/json/monsters.json [hunk 1]
```diff
@@ -1651,13 +1651,13 @@
...
     "name": "skeletal dog",
-    "description": "This canines see-through thin, black veined and scab riddled skin, is tightly compressed around its skeletal frame.",
+    "description": "This canine skeletal frame with bulging bones, woven with a net of sinew growing out from conglutinated carapace of fused together ribs, resembles an overactive turtle.",
     "default_faction": "zombie",
     "species": [ "ZOMBIE" ],
     "diff": 12,
     "size": "SMALL",
     "hp": 12,
-    "speed": 120,
+    "speed": 100,
```
Comment (Night-Pryanik) [inline data/json/monsters.json]: Why the decrease?

Comment (nexusmrsep) [reply data/json/monsters.json]: As stated in  #23463 the overgrown bone carapace and reduced muscle mass would reduce speed.

**Comment by DracoGriffin** [comment]
Both descriptions are both really awkward and stilited, I like where you're going with it but it's difficult to follow. Can you come up with a few more examples?

Dog:
`A once-canine encased in bony armor, the massive conglutinated carapace seems fueled by a network of black-filled veins and sickly organs.`

Skeleton:
`A monstrous overgrowth of ossified tissue has replaced this zombie's rotting skin with an organic armor of dense bone.  Large clumps of black goo seep from its joints, as it makes sickening cracks while it moves unfazed.`

**Comment by nexusmrsep** [comment]
Yeah, all suggestions are welcomed as I'm non a native speaker. 

Dog, skeletal:
Original proposition:
`This canine skeletal frame with bulging bones, woven with a net of sinew growing out from conglutinated carapace of fused together ribs, resembles an overactive turtle.`

Variant 1:
`This once-canine has shed all its' skin revealing a carapace of fused bones and ribs. This bone armour devoid of external flesh seems to be propelled by a net of sickly black veins and sinews glistening with black goo.`

Variant 2:
`This undead canine has lost it's skin and most of it's organs, but that what's left is now hidden under thick bone carapace grown all over it. Ropes of black veins and sinews oozing with black goo originate from the fused ribcage and connect to lean muscles hidden in bony shell composing it's limbs.`

Skeleton:
Original proposition:
`On a closer look it is not a decrepit zombie you expected. While still having a general look of a human skeleton, bones have regrown over some muscles and conglutineted ribs created a protective carapace. Few gaps in this grim exoskeleton show a glistening black mass of transformed organs and muscles, that controls an excesive net of sinews woven over and throughout the bones, keeping this monster mobile.`

Variant 1:
`Like a shell of a hermit crab, this skeleton has become a host to undead unknown, that accommodated this once living body into a walking suit of bone armour. Hiding inside conglutinated carapace that once was a ribcage it controls rest of the petrified members by a net of sickly black veins and sinews glistening with black goo.`

Variant 2
`This once-human creature has lost it's skin and most of it's organs, but that what's left is now hidden under thick bone carapace grown all over it. Ropes of black veins and sinews oozing with black goo originate from the fused ribcage and connect to lean muscles hidden in bony shell composing it's limbs.`

Feel free to discuss, vote for one of the variants or propose another.

**Comment by Rivet-the-Zombie** [comment]
it's -> contraction of 'it is'
its -> possessive form of 'it'

**Comment by DracoGriffin** [comment]
Honestly, after the variants, I think I like my examples more. Otherwise, variant 1 for canine, variant 2 for skeleton.

**Comment by nexusmrsep** [comment]
Ok, so after adding mostly everything in the source issue #23463 I have some questions before I flag this PR as READY:

1. Should I add any other skeletal monster fitting between skeleton (human) and skeletal juggernaut? Like skeletal brute or something?

2. Should skeletal dogs evolve into any other skeletal monsters, perhaps even juggernauts?

3. Should HARDTOSHOOT flag reduce the ranged profile of TINY monsters? Now it reduces size of monsters by one for the sake of ranged combat calculations, but it does not reduce it beyond TINY. Howewer code allows to reduce it further, so I ask if my approach is ok.

4. Are the skeletal juggernaut's stats acceptable? It is a slow monster, dragging it's bony prison-armour around, but heavily armoured, slender (HARDTOHIT) but with a heavy punch when it comes to it. Consider it a skeletal version of a zombie hulk, with a twist. It burns to a zombie fiend, which as a matter of fact fits the theme idealy.

**Comment by kevingranade** [comment]
1. Would be nice if you feel like it, but can also be added later.
2. Is certainly possible, though I'd rather it be a distinct monster, quadruped vs biped is a pretty big change to ignore.
3. It can, but probably isn't necessary.
4. Looks ok to me, if it needs tweakingbit can happen later.

**Comment by kevingranade** [comment]
I'm testing this right now, and just FYI:
HARDTOHIT is working as intended, they're hard to hit, but it gives good feedback.
Regular skeleton:
  9mm ammo can't damage them at all.
  It takes a number of rounds of .45 (3-5) to put one down.
  Entry level spears can trivally chip them down to nothing.
Skeletal Juggernaut:
  Trivial to kite with a gun.
  Can chip away at HP with a .45 (I'd like to increase the armor to get it out of range of most handguns, but not necessary for this PR), but it takes a LOT of ammo.
  Can take down with an AKM with ~15 rounds, that's a significant investment IMO.
  Can't safely kite with a reach weapon due to the reach attack, good.

**Comment by kevingranade** [comment]
Bug: when skeleton armor negates damage, it doesn't print what the round bounces off of.

**Pull request merged**